packages/capacitor-plugin/ios/Sources/GeolocationPlugin/GeolocationPlugin.swift

```diff
@@ -0,0 +1,167 @@
+import Capacitor
+import OSGeolocationLib
+
+import Combine
+
+@objc(GeolocationPlugin)
+public class GeolocationPlugin: CAPPlugin, CAPBridgedPlugin {
+    public let identifier = "GeolocationPlugin"
+    public let jsName = "GeolocationPlugin"
+    public let pluginMethods: [CAPPluginMethod] = [
+        .init(name: "getCurrentPosition", returnType: CAPPluginReturnPromise),
+        .init(name: "watchPosition", returnType: CAPPluginReturnCallback),
+        .init(name: "clearWatch", returnType: CAPPluginReturnPromise),
+        .init(name: "checkPermissions", returnType: CAPPluginReturnPromise),
+        .init(name: "requestPermissions", returnType: CAPPluginReturnPromise)
+    ]
+
+    private var plugin: (any OSGLOCService)?
+    private var cancellables = Set<AnyCancellable>()
+    private var callbackManager: GeolocationCallbackManager?
+    private var isInitialised: Bool = false
+
+    public override func load() {
+        self.plugin = OSGLOCManagerWrapper()
```

Comment:
Not super important, but should this variable be called `plugin`? Would it make more sense to name it something like `locationManager` or `locManagerWrapper`? When I see `plugin` I think of the actual bridge, the `GeolocationPlugin` class.

Reply:
This was a copy-paste from InAppBrowser but I can change it.